src/System.Threading/tests/AsyncLocalTests.cs

```diff
@@ -419,5 +423,162 @@ private static Task Run(Func<Task> func)
+                        {
+                            if (index >= i)
+                            {
+                                Assert.Equal(index, (int) locals[index].Value);
+                            }
+                            else
+                            {
+                                Assert.Null(locals[index].Value);
+                            }
+                        }
+
+                        ExecutionContext.Run(Default.CreateCopy(), _ =>
+                        {
+                            for (int index = 0; index < locals.Length; index++)
+                            {
+                                Assert.Null(locals[index].Value);
+                            }
+                        }, null);
+
+                        for (int l = 0; l < locals.Length; l++)
+                        {
+                            Assert.Equal(0, setManually[l]);
+                            Assert.Equal(l < i ? 0 : 1, unsetAutomatically[l]);
+                            Assert.Equal(l < i ? 0 : 2, setAutomatically[l]);
+                        }
+
+                        for (int c = 0; c < locals.Length; c++)
+                        {
+                            ExecutionContext.Run(contexts[c].CreateCopy(), _ =>
+                            {
+                                for (int index = locals.Length - 1; index >= 0; index--)
+                                {
+                                    if (index >= c)
+                                    {
+                                        Assert.Equal(index, (int) locals[index].Value);
+                                    }
+                                    else
+                                    {
+                                        Assert.Null(locals[index].Value);
+                                    }
+                                }
+                            }, null);
+                        }
+                    }, null);
+
+                    Array.Clear(setManually, 0, count);
+                    Array.Clear(unsetAutomatically, 0, count);
+                    Array.Clear(setAutomatically, 0, count);
+                }
+            }
+        }
+
+        static async Task AsyncRecursive(int index, AsyncLocal<object>[] locals, ExecutionContext[] contexts, Action<AsyncLocalValueChangedArgs<object>> valueChangedHandler)
+        {
+            locals[index] = new AsyncLocal<object>(valueChangedHandler);
+            locals[index].Value = index;
+
+            contexts[index] = ExecutionContext.Capture();
+
+            if (index > 0)
+            {
+                await AsyncRecursive(index - 1, locals, contexts, valueChangedHandler);
+            }
+        }
+
+        public static IEnumerable<object[]> GetCount()
+        {
+            const int max = 40;
+
+            foreach (int i in Enumerable.Range(1, max))
+            {
+                yield return new object[] { i };
+            }
+        }
```

Comment:
Might be simpler with LINQ, e.g.
```C#
public static IEnumerable<object[]> GetCounts() =>
    Enumerable.Range(0, 40).Select(i => new object[] { i });
```

Reply:
Done, though starting from 1 (for 1 asynclocal)